Source/WebCore/platform/graphics/FontCascade.cpp

```diff
@@ -1250,10 +1250,11 @@ bool FontCascade::rightExpansionOpportunity(StringView stringView, TextDirection
 
 bool FontCascade::canReceiveTextEmphasis(char32_t c)
 {
-    if (U_GET_GC_MASK(c) & (U_GC_Z_MASK | U_GC_CN_MASK | U_GC_CC_MASK | U_GC_CF_MASK))
+    if (U_GET_GC_MASK(c) & (U_GC_Z_MASK | U_GC_CN_MASK | U_GC_CC_MASK | U_GC_CF_MASK | U_GC_P_MASK))
         return false;
 
```

Comment:
```
if (U_GET_GC_MASK(c) & U_GC_P_MASK)
    return c == '#' || c == '%' || c == '&' || [...]
```

Implementing the exceptions should be something like this, but maybe @darinadler has a better idea.

I'd reuse the list from this Firefox changes: https://phabricator.services.mozilla.com/D219099

Reply:
If we have a performance problem we can make an ASCII table or a sorted array map. Generally a special case for ASCII works well, with one branch to range check and then a simple index into a 128-element array.